honeybee_radiance/modifier/material/absdf.py

```diff
@@ -0,0 +1,261 @@
+        5+ BSDFfile ux uy uz funcfile transform
+        0
+        0|3|6|9
+            rfdif gfdif bfdif
+            rbdif gbdif bbdif
+            rtdif gtdif btdif
+
+    The __init__ method sets additional diffuse reflectance for front and back as well
+    as additional diffuse transmittance to 0. You can setup these values by using their
+    respective property.
+
+    Args:
+        bsdf_file: Path to an xml file. Data will NOT be cached in memory.
+        identifier: Text string for a unique Material ID. Must not contain spaces
+            or special characters. This will be used to identify the object across
+            a model and in the exported Radiance files. If None, the identifier
+            will be derived from the bsdf_file name. (Default: None)
+        up_orientation: (x, y ,z) vector that sets the hemisphere that the
+            BSDF material faces.  For materials that are symmetrical about
+            the face plane (like non-angled venetian blinds), this can be
+            any vector that is not perfectly normal to the face. For
+            asymmetrical materials like angled venetian blinds, this variable
+            should be coordinated with the direction the face are facing.
+            The default is set to (0.01, 0.01, 1.00), which should hopefully
+            not be perpendicular to any typical face.
+        modifier: Material modifier (Default: None).
+        function_file: Optional input for function file (Default: .).
+        transform: Optional transform input to to scale the thickness and reorient
+            the up vector (default: None).
+        angle_basis: BSDF file angle basis. If not provided by user honeybee tries to
+            find it by parsing BSDF file itself.
+        dependencies: A list of primitives that this primitive depends on. This
+            argument is only useful for defining advanced primitives where the
+            primitive is defined based on other primitives. (Default: [])
+
+
+    Properties:
+        * identifier
+        * display_name
+        * bsdf_file
+        * up_orientation
+        * function_file
+        * transform
+        * angle_basis
+        * front_diffuse_reflectance
+        * back_diffuse_reflectance
+        * diffuse_transmittance
+        * dependencies
+        * values
+        * modifier
+        * dependencies
+        * is_modifier
+        * is_material
+    """
+
+    __slots__ = ()
+
+    # TODO(): compress file content: https://stackoverflow.com/a/15529390/4394669
+    def __init__(self, bsdf_file, identifier=None, up_orientation=None, modifier=None, 
+                 function_file='.', transform=None, angle_basis=None, dependencies=None):
+        """Create aBSDF material."""
+        # thickness is ignored in aBSDF class
+        thickness = None
+        BSDF.__init__(self, bsdf_file, identifier, up_orientation, thickness,
+                 modifier, function_file, transform, angle_basis,
+                 dependencies)
```

Comment:
I see. The right way of doing this is probably to use aBSDF as the base class and subclass BSDF from it. In that scenario, you can add thickness as the extra property to aBSDF. I would say we should do that if it is not too much work.